subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesKotlinDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,2252 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.testsuites.dependencies
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.dsl.GradleDsl
+import spock.lang.Ignore
+
+
+class TestSuitesKotlinDSLDependenciesIntegrationTest extends AbstractIntegrationSpec {
+    private versionCatalog = file('gradle', 'libs.versions.toml')
+
+    // region basic functionality
+    def 'suites do not share dependencies by default'() {
+        given:
+        buildKotlinFile << """
+            plugins {
+                `java-library`
+            }
+
+            ${mavenCentralRepository(GradleDsl.KOTLIN)}
+
+            testing {
+                suites {
+                    val test by getting(JvmTestSuite::class) {
```

Comment:
Personally I feel this is worse than doing:
```suggestion
                    getByName<JvmTestSuite>("test") {
```
(or `named`)
because you don't use the variable. I guess it would help with consistency, but it just looks weird if you don't use the variable.

Likewise with `by registering` vs `register`.

Reply:
It might look a little weird, but we've written literally all the Test Suites example (and much other Kotlin conversions of Groovy scripts) using this style, so I don't think we should change it.